bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/rest/api/authentication/AuthenticationApi.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.automower.internal.rest.api.authentication;
+
+import org.apache.commons.httpclient.HttpStatus;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.FormContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.util.Fields;
+import org.openhab.binding.automower.internal.rest.api.HusqvarnaApi;
+import org.openhab.binding.automower.internal.rest.api.authentication.dto.PostOAuth2Response;
+import org.openhab.binding.automower.internal.rest.exceptions.AutomowerCommunicationException;
+import org.openhab.binding.automower.internal.rest.exceptions.UnauthorizedException;
+
+import com.google.gson.JsonSyntaxException;
+
+/**
+ * Gives access to the Husqvarna authentication api
+ *
+ * @author Markus Pfleger - Initial contribution
+ */
+@NonNullByDefault
+public class AuthenticationApi extends HusqvarnaApi {
+
+    public AuthenticationApi(HttpClient httpClient) {
+        super(httpClient);
+    }
+
+    @Override
+    public String getBaseUrl() {
+        return "https://api.authentication.husqvarnagroup.dev/v1";
+    }
+
+    public PostOAuth2Response loginOAuth2(String appKey, String username, String password)
+            throws AutomowerCommunicationException {
+        final Request request = getHttpClient().newRequest(getBaseUrl() + "/oauth2/token");
+        request.method(HttpMethod.POST);
+
+        Fields fields = new Fields();
+        fields.put("grant_type", "password");
+        fields.put("client_id", appKey);
+        fields.put("username", username);
+        fields.put("password", password);
+
+        request.content(new FormContentProvider(fields));
+
+        ContentResponse response;
+
+        try {
+            response = request.send();
+        } catch (Exception e) {
```

Comment:
See above and below.
```suggestion
        } catch (InterruptedException | TimeoutException | ExecutionException e) {
```

Reply:
Will change that. I tried to do this to have a defined exception even when the 3rd party lib would throw something unexpected (null pointer or whatever). Probably the framework has some handling of such runtime exceptions and keeps my thing/bridge in a consistent state, so I will change that.

Reply:
The framework will log any uncatched exceptions. Since this would be a bug in you binding or a third-party lib, it's good to fail hard. The framework takes care that this doesn't affect itself or any other bindings.